rootless/rootless.go

```diff
@@ -23,3 +31,45 @@ func RunningWithRootlessKit() bool {
 	})
 	return runningWithRootlessKit
 }
+
+// GetRootlessKitInfo returns the Info object of RootlessKit REST API: https://github.com/rootless-containers/rootlesskit/blob/master/pkg/api/openapi.yaml
+//
+// The object is returned as a json.RawMessage, without unmarshalling into a struct of github.com/rootless-containers/rootlesskit/pkg/api.Info ,
+// so as to retain all the fields that are unknown to github.com/rootless-containers/rootlesskit/pkg/api.Info .
+func GetRootlessKitInfo(ctx context.Context) (json.RawMessage, error) {
+	stateDir := os.Getenv("ROOTLESSKIT_STATE_DIR")
+	if stateDir == "" {
+		return nil, errors.New("environment variable `ROOTLESSKIT_STATE_DIR` is not set")
+	}
+	apiSock := filepath.Join(stateDir, "api.sock")
+	if _, err := os.Stat(apiSock); err != nil {
+		return nil, err
+	}
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+				var d net.Dialer
+				return d.DialContext(ctx, "unix", apiSock)
+			},
+		},
+	}
+	req, err := http.NewRequest("GET", "http://rootlesskit/v1/info", nil)
```

Comment:
This feels like we're making an impromptu API client; could we have a basic API client in the rootlesskit repository that we can vendor and consume?

Reply:
updated to import `rootlesskit/pkg/api/client`